airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/base/IntegrationRunner.kt

```diff
@@ -339,11 +343,37 @@ internal constructor(
         }
     }
 
+    class ThreadCreationInfo {
+        val stack: List<StackTraceElement> = Thread.currentThread().stackTrace.asList()
+        val time: Instant = Instant.now()
+        override fun toString(): String {
+            return "creationStack=${stack.joinToString("\n  ")}\ncreationTime=$time"
+        }
+    }
+
     companion object {
         private val LOGGER: Logger = LoggerFactory.getLogger(IntegrationRunner::class.java)
+        private val threadCreationInfo: InheritableThreadLocal<ThreadCreationInfo> =
+            object : InheritableThreadLocal<ThreadCreationInfo>() {
+                override fun childValue(parentValue: ThreadCreationInfo): ThreadCreationInfo {
+                    return ThreadCreationInfo()
+                }
+            }
 
         const val TYPE_AND_DEDUPE_THREAD_NAME: String = "type-and-dedupe"
 
+        // ThreadLocal.get(Thread) is private. So we open it and keep a reference to the
+        // opened method
+        private val getMethod: Method =
+            ThreadLocal::class.java.getDeclaredMethod("get", Thread::class.java).also {
+                it.isAccessible = true
```

Comment:
I'm getting an error here, not sure if I'm doing something silly or if there was a bug introduced here?
```
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make private java.lang.Object java.lang.ThreadLocal.get(java.lang.Thread) accessible: module java.base does not "opens java.lang" to unnamed module @5f058f00
```

Reply:
I'm not sure how you're executing this. I've tested this when running tests from intelliJ and airbyte-ci, as well as when running a connector through the docker image

Reply:
@stephane-airbyte try running this main method through Intellij IDE: 

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-mysql/src/main/java/io/airbyte/integrations/destination/mysql/MySQLDestination.java#L188-L193

With the following parameters: 
- Java 21 SDK
- Cli arguments `--write --config /<my_path>/config.json --catalog /<my_path>/configured_catalog.json`
- Click `Modify options`  -> `Redirect input` and then add example input message `<my_path>/message.json`

Reply:
you need an additional JVM parameters, `--add-opens=java.base/java.lang=ALL-UNNAMED`

Reply:
Roger -- thanks!